### PR TITLE
RWA-2371: Spring boot upgrade.

### PR DESCRIPTION
**Pull request by adityadwadasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-2371

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmonitor/SpringBootFunctionalBaseTest.java [hunk 1]
```diff
@@ -49,6 +50,8 @@
 @SpringBootTest
 @ActiveProfiles({"functional"})
 @RunWith(SpringIntegrationSerenityRunner.class)
+//@Import({CoreCaseDataApi.class,BearerTokenGenerator.class})
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmonitor/SpringBootFunctionalBaseTest.java:53]: remove commented code?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java [hunk 2]
```diff
@@ -58,11 +57,11 @@ void givenCamundaClientIsCalledShouldReturnTasks() throws IOException {
                 null,
                 ZonedDateTime.parse(
                     "2021-06-02T16:17:32.808+0000",
-                    CAMUNDA_DATA_TIME_FORMATTER.withZone(ZoneId.of("UTC"))
+                    CAMUNDA_DATA_TIME_FORMATTER
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java:60]: Does this default to UTC?

**Review by raghera — COMMENTED** [review]
Looks good.  Just a couple of questions.

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmonitor/SpringBootFunctionalBaseTest.java (hunk 1 above)
Comment (adityadwadasi) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmonitor/SpringBootFunctionalBaseTest.java]: I think its due to the master branch merge I've got some in consistencies. I corrected it now Ravi.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java (hunk 2 above)
Comment (adityadwadasi) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java:60]: This doesn't really defaults to UTC. But it is converting the date time string to ZonedDateTime Ravi.

File: build.gradle [hunk 3]
```diff
@@ -368,15 +369,19 @@ dependencies {
   testImplementation group: 'info.solidsoft.gradle.pitest', name: 'gradle-pitest-plugin', version: '1.15.0'
   testImplementation group: 'org.codehaus.sonar-plugins', name: 'sonar-pitest-plugin', version: '0.5'
 
-  testImplementation group: 'com.github.tomakehurst', name: 'wiremock-jre8', version: '2.35.2', {
+  /*testImplementation group: 'com.github.tomakehurst', name: 'wiremock-jre8', version: '3.0.1', {
     exclude group: 'com.github.jknack.handlebars.java'
-  }
+  }*/
```
Comment (raghera) [inline build.gradle:374]: Do we need this?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java (hunk 2 above)
Comment (raghera) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java:60]: But a ZonedDateTime usually has a Zone, so which Zone is it going to use here?  Previously it was `UTC` but is it now just the Zone of the machine the test is running in?  I am just thinking of why you had to remove the Zone and if it's going to become inconsistent.

**Review by raghera — COMMENTED** [review]
Looks good but a couple of questions.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java (hunk 2 above)
Comment (raghera) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmonitor/clients/camunda/CamundaClientTest.java:60]: We discussed and looks fine.

**Review by raghera — DISMISSED** [review]

**Pull request merged**